contracts/GenericLender/GenericAave.sol

```diff
@@ -72,10 +74,16 @@ contract GenericAave is GenericLenderBase {
 
     // for the management to activate / deactivate incentives functionality
     function setIsIncentivised(bool _isIncentivised) external management {
-        require(address(aToken.getIncentivesController()) != address(0), "!aToken does not have incentives controller set up");
+        // NOTE: if the aToken is not incentivised, getIncentivesController() might revert (aToken won't implement it)
+        // to avoid calling it, we use the OR and lazy evaluation
+        require(!_isIncentivised || address(aToken.getIncentivesController()) != address(0), "!aToken does not have incentives controller set up");
```

Comment:
is there ever a need or possibility to activate incentives on a strategy/pool that didn't start incentivized?

If this is the case we may want the require to be in an if block instead
```
if (_isIncentivised) {
  require( address(aToken.getIncentivesController()) != address(0), "!aToken does not have incentives controller set up");
}  

```

Other wise i don't see how we can activate the logic once the contract is setup as false

Reply:
you are right, i was confusing the conditions